php_mysqlnd_azure.c

```diff
@@ -28,39 +28,52 @@
 
 ZEND_DECLARE_MODULE_GLOBALS(mysqlnd_azure)
 
-/* {{{ PHP_GINIT_FUNCTION */
-static PHP_GINIT_FUNCTION(mysqlnd_azure)
-{
-#if defined(COMPILE_DL_MYSQLND_AZURE) && defined(ZTS)
-	ZEND_TSRMLS_CACHE_UPDATE();
-#endif
-	mysqlnd_azure_globals->enabled = 0;
-	mysqlnd_azure_globals->redirectCache = NULL;
-}
-/* }}} */
 
-/* {{{ PHP_GSHUTDOWN_FUNCTION */
-static PHP_GSHUTDOWN_FUNCTION(mysqlnd_azure)
+/* {{{ OnUpdateEnableRedirect */
+static ZEND_INI_MH(OnUpdateEnableRedirect)
 {
-	if (mysqlnd_azure_globals->redirectCache) {
-		zend_hash_destroy(mysqlnd_azure_globals->redirectCache);
-		mnd_pefree(mysqlnd_azure_globals->redirectCache, 1);
-		mysqlnd_azure_globals->redirectCache = NULL;
-	}
+    if ((ZSTR_LEN(new_value) == 9 && strcasecmp("preferred", ZSTR_VAL(new_value)) == 0)
+        || (ZSTR_LEN(new_value) == 1 && strcasecmp("2", ZSTR_VAL(new_value)) == 0)) {
+
+        MYSQLND_AZURE_G(enableRedirect) = REDIRECT_PREFERRED;
+
+    } else if ((ZSTR_LEN(new_value) == 2 && strcasecmp("on", ZSTR_VAL(new_value)) == 0)
+        || (ZSTR_LEN(new_value) == 3 && strcasecmp("yes", ZSTR_VAL(new_value)) == 0)
+        || (ZSTR_LEN(new_value) == 4 && strcasecmp("true", ZSTR_VAL(new_value)) == 0)
+        || (ZSTR_LEN(new_value) == 1 && strcasecmp("1", ZSTR_VAL(new_value)) == 0)) {
```

Comment:
Since `zend_ini_parse_bool()` is indeed only available as of PHP 7.3.0 (I wasn't aware of that), this is the way to go. A possible improvement might be to encapsulate each comparison in a macro, to make the code more readable and easier to maintain. Could read something like `STRING_EQUALS(new_value, "1")` then.

Reply:
Hi Christoph, I've made the change and added it in another separate PR without cache change which has been finished. Thanks a lot for your suggestion.